src/cmd.rs

```diff
@@ -82,28 +82,28 @@ pub trait Adxl345Reader {
     /// ## Arguments
     /// * `register` - Register address to be accessed (read).
     ///
-    fn access(&self, register: u8) -> AdxlResult<u8>;
+    fn access(&mut self, register: u8) -> AdxlResult<u8>;
```

Comment:
Question I have is why this needs to be changed to &mut self for something that shouldn't be modifying anything just reading from the device

Reply:
The embedded-hal trait defines read method(s) as `&mut self`.

And I think that actually does make sense. This read is not a simple memory read. It is a device access. Which does in fact involve writes. Both, to data structures in the operating system interfaces and to the device memory itself.

Therefore, it doesn't make sense for I2C write to be `&mut self` and I2C read to be `&self`.
If one would want to model this with interior mutability and proper synchronization, then both I2C read and write should be `&self`.

In fact, now that I think about it, having `&self` read methods probably is unsound.
One could have two threads that access the device simultaneously, by sending two clones of `Arc<Device>` to two threads. That would cause memory safety issues at the I2C operating system interface level.